tests/usbredir_test.go

```diff
@@ -83,100 +84,139 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 				select {
-				case response := <-readStop:
-					By("Checking the response from QEMU's usbredir")
-					// Comparing the actual messages could be error prone due the fact that:
-					// 1. Part of the return value is a qemu release version, e.g: 5.2.0 (test would break with different release!)
-					// 2. Capabilities can change over time which means the message would be different then the one hardcoded, correct nonetheless.
-					// I'm keeping the helloMessageRemote to have a proof of working example that could also be used if needed.
-					Expect(response).ToNot(BeEmpty(), "response should not be empty")
-					Expect(response).To(HaveLen(len(helloMessageRemote)))
-				case err = <-k8ResChan:
-					Expect(err).ToNot(HaveOccurred())
-				case <-time.After(45 * time.Second):
-					Fail("Timout reached while waiting for valid response")
-				case <-connStop:
-					return
+				case err := <-errors[i]:
+					Expect(err).To(MatchError(ContainSubstring("websocket: bad handshake")))
+					Expect(i).To(Equal(v1.UsbClientPassthroughMaxNumberOf))
+				case <-time.After(time.Second):
+					stops[i] <- struct{}{}
+					Expect(i).ToNot(Equal(v1.UsbClientPassthroughMaxNumberOf))
 				}
 			}
+		})
+
+		It("Should work in parallel", func() {
+			stops := make([]chan struct{}, v1.UsbClientPassthroughMaxNumberOf)
+			errors := make([]chan error, v1.UsbClientPassthroughMaxNumberOf)
+			for i := 0; i < v1.UsbClientPassthroughMaxNumberOf; i++ {
+				stops[i] = make(chan struct{})
+				errors[i] = make(chan error)
+				go runConnectGoroutine(virtClient, vmi, stops[i], errors[i])
+				// avoid too fast requests which might get denied by server (to avoid flakyness)
+				time.Sleep(100 * time.Millisecond)
+			}
 
-			It("Should work several times", func() {
-				for i := 0; i < 10; i++ {
-					usbredirConnect(nil)
+			for i := 0; i < v1.UsbClientPassthroughMaxNumberOf; i++ {
+				select {
+				case err := <-errors[i]:
+					Expect(err).ToNot(HaveOccurred())
+				case <-time.After(time.Second):
+					stops[i] <- struct{}{}
 				}
-			})
-
-			It("Should work in parallel", func() {
-				connStop := make(chan struct{})
-				defer close(connStop)
-				for i := 0; i < v1.UsbClientPassthroughMaxNumberOf; i++ {
-					go func() {
-						defer GinkgoRecover()
-						usbredirConnect(connStop)
-					}()
+			}
+		})
+
+		It("Should work several times", func() {
+			for i := 0; i < 4*v1.UsbClientPassthroughMaxNumberOf; i++ {
+				stop := make(chan struct{})
+				errch := make(chan error)
+				go runConnectGoroutine(virtClient, vmi, stop, errch)
+
+				select {
+				case err := <-errch:
+					Expect(err).ToNot(HaveOccurred())
+				case <-time.After(time.Second):
+					stop <- struct{}{}
 				}
-				// Wait for connection, write and read of all above.
-				time.Sleep(5 * time.Second)
-			})
+			}
 		})
 	})
 })
 
+func runConnectGoroutine(
+	virtClient kubecli.KubevirtClient,
+	vmi *v1.VirtualMachineInstance,
+	stop chan struct{},
+	errch chan error,
+) {
+	defer GinkgoRecover()
```

Comment:
Should this panic?

Reply:
When I wrote the code it did and I kept it. Should it be removed? I'll double check if this is needed or not.

Reply:
Removed.